src/main/java/Duke/Parser.java

```diff
@@ -0,0 +1,133 @@
+package Duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import Duke.Command.AddCommand;
+import Duke.Command.Command;
+import Duke.Command.DelCommand;
+import Duke.Command.ExitCommand;
+import Duke.Command.FindCommand;
+import Duke.Command.InvalidCommandException;
+import Duke.Command.ListCommand;
+import Duke.Command.MarkCommand;
+import Duke.Command.UnmarkCommand;
+import Duke.Task.DeadlineTask;
+import Duke.Task.EventTask;
+import Duke.Task.ToDoTask;
+
+public class Parser {
+
+    /** The datetime format for user input. */
+    public static final DateTimeFormatter DATETIME_FORMATTER =
+            DateTimeFormatter.ofPattern("d.M.yyyy HH:mm");
+
+    /** 
+     * The error string to be displayed when the date time format provided by the
+     * is wrong.
+     */
+    private static final String INVALID_TIME_FORMAT_MSG = 
+            "☹ OOPS!!! Please use the following format for time dd.mm.yyyy hh:mm";
+
+    /** 
+     * The error string to be displayed when the number format provided by the
+     * is wrong.
+     */
+    private static final String INVALID_NUMBER_FORMAT_MSG = 
+            "☹ OOPS!!! Please use the following format for time dd.mm.yyyy hh:mm";
+
+    /**
+     * Parse the user input into a command.
+     * 
+     * @param input The user input
+     * @return
+     * @throws InvalidCommandException
+     */
+    public static Command parse(String input) throws InvalidCommandException {
+        String errMsg = "";
+        if (input.startsWith("bye")) {
+            return new ExitCommand();
+        } else if (input.startsWith("list")) {
+            return new ListCommand();
+        } else if (input.startsWith("mark")) {
```

Comment:
Would it be better to use a switch statement here? Not sure if that would be cleaner